src/serac/physics/heat_transfer.hpp

```diff
@@ -522,26 +522,37 @@ class HeatTransfer<order, dim, Parameters<parameter_space...>, std::integer_sequ
   /**
    * @brief Solve the adjoint problem
    * @pre It is expected that the forward analysis is complete and the current temperature state is valid
-   * @note If the essential boundary state is not specified, homogeneous essential boundary conditions are applied
+   * @pre The adjoint load maps are expected to contain a single entry named "temperature"
+   * @note If the essential boundary dual is not specified, homogeneous essential boundary conditions are applied to
+   * the adjoint system
    *
-   * @param[in] adjoint_load The dual state that contains the right hand side of the adjoint system (d quantity of
-   * interest/d temperature)
-   * @param[in] dual_with_essential_boundary A optional finite element dual containing the non-homogenous essential
-   * boundary condition data for the adjoint problem
-   * @return The computed adjoint finite element state
+   * @param adjoint_loads An unordered map containing finite element duals representing the RHS of the adjoint equations
+   * indexed by their name
+   * @param duals_with_essential_boundary A unordered map containing finite element duals representing the
+   * non-homogenous essential boundary condition data for the adjoint problem indexed their name
```

Comment:
```suggestion
   * @param duals_with_essential_boundary An unordered map containing finite element duals representing the
   * non-homogenous essential boundary condition data for the adjoint problem indexed by their name
```
I can't remember if it's "homogeneous" or "homogenous".

On a side note: what is this argument (`duals_with_essential_boundary`) for? If it relates to the essential boundary condition values for a given time, should we be asking the user for that information or just keep track of it ourselves? If it is just the values of the essential bcs, should it be a dual or a primal field?

Reply:
The adjoint system can have non-zero essential boundary conditions if the boundary values are part of the quantity of interest. This is common for parameter estimation in inverse problems where the quantity of interest is a particular experimental data point. You are correct that I believe this should be a `FiniteElementState` instead of a `FiniteElementDual`.